src/soot/LambdaMetaFactory.java

```diff
@@ -0,0 +1,272 @@
+            return null;
+        }
+        String samMethodType = ((ClassConstant) bootstrapArgs.get(0)).getValue();
+        SootMethodRef implMethod = ((MethodHandle) bootstrapArgs.get(1)).getMethodRef();
+        String instantiatedMethodType = ((ClassConstant) bootstrapArgs.get(2)).getValue();
+        if(!samMethodType.equals(instantiatedMethodType)) {
+            G.v().out.println("warning: LambdaMetaFactory: " + 
+                              samMethodType + " != " + instantiatedMethodType);
+            return null;
+        }
+        int flags = 0;
+        if(bootstrapArgs.size() > 3)
+            flags = ((IntConstant) bootstrapArgs.get(3)).value;
+        boolean serializable = (flags & 1 /* FLAGS_SERIALIZABLE */) != 0;
+        List<ClassConstant> markerInterfaces = new ArrayList<ClassConstant>();
+        List<ClassConstant> bridges = new ArrayList<ClassConstant>();
+        int va = 4;
+        if((flags & 2 /* FLAG_MARKERS */) != 0) {
+            if(va == bootstrapArgs.size() || !(bootstrapArgs.get(va) instanceof IntConstant)) {
+                G.v().out.println("warning: LambdaMetaFactory: unexpected arguments for LambdaMetaFactory.altMetaFactory");
+                return null;
+            }
+            int count = ((IntConstant) bootstrapArgs.get(va++)).value;
+            for(int i=0;i<count;i++) {
+                if(va >= bootstrapArgs.size()) {
+                    G.v().out.println("warning: LambdaMetaFactory: unexpected arguments for LambdaMetaFactory.altMetaFactory");
+                    return null;
+                }
+                Value v = bootstrapArgs.get(va++);
+                if(!(v instanceof ClassConstant)) {
+                    G.v().out.println("warning: LambdaMetaFactory: unexpected arguments for LambdaMetaFactory.altMetaFactory");
+                    return null;
+                }
+                markerInterfaces.add((ClassConstant)v);
+            }
+        }
+        if((flags & 4 /* FLAG_BRIDGES */) != 0) {
+            if(va == bootstrapArgs.size() || !(bootstrapArgs.get(va) instanceof IntConstant)) {
+                G.v().out.println("warning: LambdaMetaFactory: unexpected arguments for LambdaMetaFactory.altMetaFactory");
+                return null;
+            }
+            int count = ((IntConstant) bootstrapArgs.get(va++)).value;
+            for(int i=0;i<count;i++) {
+                if(va >= bootstrapArgs.size()) {
+                    G.v().out.println("warning: LambdaMetaFactory: unexpected arguments for LambdaMetaFactory.altMetaFactory");
+                    return null;
+                }
+                Value v = bootstrapArgs.get(va++);
+                if(!(v instanceof ClassConstant)) {
+                    G.v().out.println("warning: LambdaMetaFactory: unexpected arguments for LambdaMetaFactory.altMetaFactory");
+                    return null;
+                }
+                bridges.add((ClassConstant)v);
```

Comment:
You gather bridging types here but never use them. Is it planned to extend the factory by the corresponding bridge methods?

Reply:
Yes (support for altMetaFactory was a little hurried and mostly just wanted to get it to parse these without blowing up). Let me revisit this and add the bridge methods.

Reply:
b28f85d (now in the pull request, below) adds support for the bridge methods